packages/ketcher-react/src/script/editor/tool/paste.ts

```diff
@@ -14,23 +14,40 @@
  * limitations under the License.
  ***************************************************************************/
 
-import { fromPaste, getHoverToFuse, getItemsToFuse, Struct } from 'ketcher-core'
+import {
+  fromItemsFuse,
+  fromPaste,
+  fromTemplateOnAtom,
+  getHoverToFuse,
+  getItemsToFuse,
+  SGroup,
+  Struct,
+  Vec2
+} from 'ketcher-core'
 import Editor from '../Editor'
 import { dropAndMerge } from './helper/dropAndMerge'
 import { getGroupIdsFromItemArrays } from './helper/getGroupIdsFromItems'
 import { getMergeItems } from './helper/getMergeItems'
+import utils from '../shared/utils'
 
 class PasteTool {
   editor: Editor
   struct: Struct
   action: any
+  templateAction: any
+  dragCtx: any
+  findItems: string[]
   mergeItems: any
+  isSingleContractedGroup: boolean
 
   constructor(editor, struct) {
     this.editor = editor
     this.editor.selection(null)
     this.struct = struct
 
+    this.isSingleContractedGroup =
+      struct.isSingleGroup() && !struct.functionalGroups.get(0).isExpanded
```

Comment:
Hmm, are you sure we can rely on `0` ?
I guess, in this case we need to get first FG by something like `struct.functionaGroups.values().next()`

Reply:
I am sure.
If id is something else that `0`, it can be considered a bug and fixed.
This "struct" is considered to be a brand-new structure with ids different from actual canvas.
And if first (and only, as we verify beforehand) group has non-0 id, it's a clear marker of a bug.